src/modules/evaluateExpression/evaluateExpressionTestData.ts

```diff
@@ -0,0 +1,708 @@
+    {
+      value: '^[A-Za-z0-9.]+@[A-Za-z0-9]+\\.[A-Za-z0-9.]+$',
+    },
+  ],
+};
+
+testData.REGEX_check_email_false = {
+  operator: 'REGEX',
+  children: [
+    {
+      value: 'carl@sussol$net',
+    },
+    {
+      value: '^[A-Za-z0-9.]+@[A-Za-z0-9]+\\.[A-Za-z0-9.]+$',
+    },
+  ],
+};
+
+// Return User or Form values
+
+testData.user = {
+  id: 2,
+  firstName: 'Carl',
+  lastName: 'Smith',
+  title: 'Import Manager',
+};
+
+testData.organisation = {
+  id: 1,
+  name: 'XYZ Pharmaceuticals',
+  category: 'Manufacturers',
+};
+
+testData.form = {
+  q1: 'Drug Registration',
+  q2: 'A',
+  q3: undefined,
+  q4: 'Panadol',
+};
+
+testData.application = {
+  id: 1,
+  name: 'Drug Registration',
+  status: 'Draft',
+  stage: 1,
+};
+
+testData.singleUserProperty = {
+  operator: 'objectProperties',
+  children: [
+    {
+      value: { object: 'user', property: 'firstName' },
+    },
+  ],
+};
+
+// SQL operator
+
+testData.getApplicationName = {
+  type: 'string',
+  operator: 'pgSQL',
+  children: [
+    {
+      value: 'SELECT name FROM application WHERE template_id = $1',
+    },
+    {
+      value: 2,
+    },
+  ],
+};
+
+testData.getListOfTemplates = {
+  type: 'array',
+  operator: 'pgSQL',
+  children: [
+    {
+      value: 'SELECT template_name FROM template',
+    },
+  ],
+};
+
+testData.countTemplates = {
+  type: 'number',
+  operator: 'pgSQL',
+  children: [
+    {
+      value: 'SELECT count(*) FROM template',
+    },
+  ],
+};
+
+testData.getListOfTemplates_noType = {
+  operator: 'pgSQL',
+  children: [
+    {
+      value: 'SELECT template_name FROM template',
+    },
+  ],
+};
+
+// GraphQL operator -- TO DO
+
+// More complex combinations
+
+testData.concatFirstAndLastNames = {
+  type: 'string',
+  operator: 'CONCAT',
+  children: [
+    {
+      operator: 'objectProperties',
+      children: [{ value: { object: 'user', property: 'firstName' } }],
+    },
+    {
+      value: ' ',
+    },
+    {
+      operator: 'objectProperties',
+      children: [{ value: { object: 'user', property: 'lastName' } }],
+    },
+  ],
+};
+
+testData.complex1 = {
+  operator: 'AND',
+  children: [
+    {
+      operator: '=',
+      children: [
+        {
+          operator: 'objectProperties',
+          children: [
+            {
+              value: { object: 'form', property: 'q1' },
+            },
+          ],
+        },
+        {
+          value: 'Drug Registration',
+        },
+      ],
+    },
+    {
+      operator: '!=',
+      children: [
+        {
+          type: 'number',
+          operator: 'pgSQL',
+          children: [
+            { value: 'SELECT COUNT(*) FROM user_organisation WHERE user_id = $1' },
+            {
+              operator: 'objectProperties',
+              children: [{ value: { object: 'user', property: 'id' } }],
+            },
+            {
+              operator: 'objectProperties',
+              children: [{ value: { object: 'organisation', property: 'id' } }],
+            },
```

Comment:
I don't think this object is in use on this query

Reply:
The test that uses this object is currently commented out, since it uses the as-yet-unimplemented GraphQL operator. But it will use this object -- I just haven't written the test query properly yet.